master/buildbot/newsfragments/build-canStartWithWorkerForBuilder.removal

```diff
@@ -0,0 +1 @@
+Build.canStartWithWorkerForBuilder static method has been made private.
```

Comment:
```suggestion
Build.canStartWithWorkerForBuilder static method has been made private and renamed _canAcquireLocks
```

Reply:
Build class can be changed via BuilderConfig's factory, so it is not cool to deprecate it.
I admit this would be a very deep hack, and most probably this change won't impact anybody

Reply:
Fixed now.